src/lib/components/Table.svelte

```diff
@@ -64,12 +69,70 @@
             selectedItems = selectedItems.filter((_, i) => i !== index);
         }
     }
+
+    async function checkScrollable() {
+        await tick();
+        if (scrollingDiv) {
+            isRightScrollable =
+                scrollingDiv.scrollWidth > scrollingDiv.clientWidth &&
+                scrollingDiv.scrollLeft < scrollingDiv.scrollWidth - scrollingDiv.clientWidth;
+            isLeftScrollable = scrollingDiv.scrollLeft > 0;
+            isTopScrollable = scrollingDiv.scrollTop > 0;
+            isBottomScrollable =
+                scrollingDiv.scrollHeight > scrollingDiv.clientHeight &&
+                scrollingDiv.scrollTop < scrollingDiv.scrollHeight - scrollingDiv.clientHeight;
+        }
+    }
+
+    $: scrollingDiv && items && checkScrollable();
+
+    function calculateScrollShadows({
+        isRightScrollable,
+        isLeftScrollable,
+        isTopScrollable,
+        isBottomScrollable,
+    }: {
+        isRightScrollable?: boolean;
+        isLeftScrollable?: boolean;
+        isTopScrollable?: boolean;
+        isBottomScrollable?: boolean;
+    }): string {
+        const shadows = [];
+        if (isRightScrollable) shadows.push('inset -5px 0 5px -5px rgba(0,0,0,0.3)');
+        if (isLeftScrollable) shadows.push('inset 5px 0 5px -5px rgba(0,0,0,0.3)');
+        if (isTopScrollable) shadows.push('inset 0 5px 5px -5px rgba(0,0,0,0.3)');
+        if (isBottomScrollable) shadows.push('inset 0 -5px 5px -5px rgba(0,0,0,0.3)');
+        return shadows.join(', ');
+    }
+
+    onMount(() => {
+        window.addEventListener('resize', checkScrollable);
+        scrollingDiv?.addEventListener('scroll', checkScrollable);
+        return () => {
+            window.removeEventListener('resize', checkScrollable);
+            scrollingDiv?.removeEventListener('scroll', checkScrollable);
+        };
+    });
 </script>
 
-<div class="flex max-h-full flex-1 flex-col overflow-y-hidden {$$props.class}">
+<div class="relative flex max-h-full w-full flex-col overflow-x-hidden overflow-y-hidden {$$props.class}">
+    <div
+        class="pointer-events-none absolute inset-0 top-[2.78rem] z-10"
+        style="box-shadow:{calculateScrollShadows({
+            isTopScrollable,
+        })}"
+    ></div>
+    <div
+        class="pointer-events-none absolute inset-0 z-10 rounded-md"
+        style="box-shadow:{calculateScrollShadows({
+            isLeftScrollable,
+            isRightScrollable,
+            isBottomScrollable,
+        })}"
+    ></div>
```

Comment:
This adds a small inset shadow to the sides of the table when they can be scrolled. Most important for the x-axis since with these larger tables it's not always obvious you can scroll to the right.